lib/deb/s3/lock.rb

```diff
@@ -52,7 +103,9 @@ def current(codename, component = nil, architecture = nil, cache_control = nil)
 
     private
     def lock_path(codename, component = nil, architecture = nil, cache_control = nil)
-      "dists/#{codename}/#{component}/binary-#{architecture}/lockfile"
+      # we need to share lock object across architectures as "architecture=all" will
+      # iterate over all existing architectures.
+      "dists/#{codename}/#{component}/binary-all/lockfile"
```

Comment:
Hmm, sharing lock object across architectures is not actually sufficient to prevent race because `Release` is shared across multiple components as well. We'll need to acquire lock object at `codename` 🤔